src/python/pants/bsp/util_rules/targets.py

```diff
@@ -110,6 +110,21 @@ async def materialize_bsp_build_targets(bsp_goal: BSPGoal) -> BSPBuildTargets:
     return BSPBuildTargets(FrozenDict(addr_specs))
 
 
+@rule
+async def resolve_bsp_build_target_identifier(
+    bsp_target_id: BuildTargetIdentifier, bsp_build_targets: BSPBuildTargets
+) -> BSPBuildTargetInternal:
+    scheme, _, target_name = bsp_target_id.uri.partition(":")
+    if scheme != "pants":
+        raise ValueError(f"Unknown BSP scheme `{scheme}` for BSP target ID `{bsp_target_id}.")
+
+    target_internal = bsp_build_targets.targets_mapping.get(target_name)
+    if not target_internal:
+        raise ValueError(f"Unknown BSP target name: {target_name}")
+
+    return target_internal
```

Comment:
It seems like this could be a method of the `BSPBuildTargets` instead, unless you expect it to need to block at some point?

Reply:
I wasn't thinking that far ahead. Seemed nicer to have call sites request a conversion of `BuildTargetIdentifier` -> `BSPBuildTargetInternal` without having to know anything at those call sites of `BSPBuildTargets`.